syntax = "proto3";

package search;

service SearchService {
  rpc Search (SearchRequest) returns (SearchResponse) {}
}

message SearchRequest {
  string input_query = 1;
}

message SearchResponse {
  repeated SearchItem items = 1;
  string answer = 2;
}

message SearchItem {
  string canister_id = 1;
  string description = 2;
  string host = 3;
  string link = 4;
  string logo = 5;
  string token_name = 6;
  string token_symbol = 7;
  string user_id = 8;
  string created_at = 9;
}